### PR TITLE
Add metrics to track leader transition time

### LINKED ISSUE #7433: Add a new metric to observe time until Leader starts processing
**Description**

With the ZEP https://github.com/zeebe-io/enhancements/pull/16 we want to improve the time between ZeebePartition get the Leader Role update and new Leader is able to start processing.

leaderTransition

### PR DESCRIPTION
**Pull request by deepthidevaki**
## Description

The new metrics calculates the duration from when ZeebePartition receives the leader role change event from raft, until all the services for the leader is installed. Only after the services are installed the commandApi starts accepting requests for that partition. The metrics covers the duration until this point.

## Related issues

closes #7433 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [x] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

**Comment by deepthidevaki** [comment]
New panel in grafana looks as follows:
image

File: monitor/grafana/zeebe.json [hunk 1]
```diff
@@ -9539,16 +9539,17 @@
...
-          "pluginVersion": "7.0.3",
+          "pluginVersion": "7.4.5",
           "repeat": "pod",
           "repeatDirection": "h",
           "scopedVars": {
             "pod": {
               "selected": false,
-              "text": "medic-cw-20-f8b1a83e1-benchmark-zeebe-0",
-              "value": "medic-cw-20-f8b1a83e1-benchmark-zeebe-0"
+              "text": "dd-leader-latency-zeebe-0",
```
Comment (Zelldon) [inline monitor/grafana/zeebe.json:9551]: I think next time would be better if you can reset the namespace variable :)

**Review by ChrisKujawa — APPROVED** [review]
Awesome thanks @deepthidevaki 

Now it contains the exporter opening as well, which I first thought would be not necessary/ideal, but I hope this should be fine. There is not much happening (just seek to a position). Plus, it is exactly the point before we open the command api for new commands so this is actually quite nice place :+1:

File: broker/src/main/java/io/camunda/zeebe/broker/system/partitions/RoleMetrics.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.broker.system.partitions;
+
+import io.prometheus.client.Gauge;
+
+public class RoleMetrics {
+  private static final Gauge LEADER_TRANSITION_LATENCY =
```
Comment (Zelldon) [inline broker/src/main/java/io/camunda/zeebe/broker/system/partitions/RoleMetrics.java:13]: We could name it also LEADER_SERVICES_READY_LATENCY

File: monitor/grafana/zeebe.json (hunk 1 above)
Comment (deepthidevaki) [reply monitor/grafana/zeebe.json:9551]: Well, I tried to do that by recreating the panel on a fresh dashboard. But the diff was much bigger. No idea why. Grafana dashboard is a mystery :smile:

**Comment by deepthidevaki** [comment]
> Now it contains the exporter opening as well, which I first thought would be not necessary/ideal, but I hope this should be fine. There is not much happening (just seek to a position). Plus, it is exactly the point before we open the command api for new commands so this is actually quite nice place 

I was also skeptic there. But I think, if we need we can also add more fine grained metrics for each step in the install process. But currently the significant time is for stream processor recovery. Others combined is only a few milliseconds.

**Pull request merged**